memutils/memory/cuda_malloc.cu

```diff
@@ -141,17 +150,57 @@ extern "C" CGoCallResHandle GetDeviceGlobalMemoryInMB(int device) {
   return resHandle;
 }
 
-extern "C" CGoCallResHandle CudaProfilerStart() {
+CGoCallResHandle CudaProfilerStart() {
   CGoCallResHandle resHandle = {NULL, NULL};
   cudaProfilerStart();
   resHandle.pStrErr = checkCUDAError("cudaProfilerStart");
   return resHandle;
 }
 
-extern "C" CGoCallResHandle CudaProfilerStop() {
+CGoCallResHandle CudaProfilerStop() {
   CGoCallResHandle resHandle = {NULL, NULL};
   cudaDeviceSynchronize();
   cudaProfilerStop();
   resHandle.pStrErr = checkCUDAError("cudaProfilerStop");
   return resHandle;
 }
+
+CGoCallResHandle GetDeviceMemoryInfo(size_t *freeSize, size_t *totalSize,
```

Comment:
we can return device level memory info as well for non-rmm managed

Reply:
this cuda_malloc.cu will be deprecated after rmm integration is proven. Before that, may be it's better to rely on our memory tracking on golang side.